contracts/StandardBounty.sol

```diff
@@ -0,0 +1,319 @@
+    uint newBalanceMult = _balance * _numerator;
+    require(newBalanceMult / _numerator == _balance);
+
+    //secondly divides by the denomenator
+    newBalanceDiv = newBalanceMult / _denomenator;
+  }
+
+  function acceptFulfillment(uint _fulfillmentId, StandardToken[] _payoutTokens, uint[] _tokenAmounts)
+      public
+      validateFulfillmentArrayIndex(_fulfillmentId)
+      sameLength(_payoutTokens.length, _tokenAmounts.length)
+      onlyController
+  {
+      hasPaidOut = true;
+      Fulfillment storage fulfillment = fulfillments[_fulfillmentId];
+      for (uint256 i = 0; i < _payoutTokens.length; i++){
+        for (uint256 j = 0; j < fulfillment.fulfillers.length; j++){
+          if (_payoutTokens[i] == address(0)){
+            require(this.balance >= _tokenAmounts[i]);
+            fulfillment.fulfillers[j].transfer(calculateFraction(_tokenAmounts[i], fulfillment.numerators[j], fulfillment.denomenator));
+          } else {
+            require(_payoutTokens[i].balanceOf(this) >= _tokenAmounts[i]);
+            require(_payoutTokens[i].transfer(fulfillment.fulfillers[j], calculateFraction(_tokenAmounts[i], fulfillment.numerators[j], fulfillment.denomenator)));
+          }
+        }
+      }
+
+      FulfillmentAccepted(_fulfillmentId);
+  }
+
+  function acceptAndFulfill(address[] _fulfillers, uint[] _numerators, uint _denomenator, string _data, StandardToken[] _payoutTokens, uint[] _tokenAmounts)
+      public
+      sameLength(_payoutTokens.length, _tokenAmounts.length)
+      onlyController
+  {
+      fulfillBounty(_fulfillers, _numerators, _denomenator, _data);
+      hasPaidOut = true;
+      Fulfillment storage fulfillment = fulfillments[fulfillments.length - 1];
+      for (uint256 i = 0; i < _payoutTokens.length; i++){
+        for (uint256 j = 0; j < fulfillment.fulfillers.length; j++){
+          if (_payoutTokens[i] == address(0)){
+            require(this.balance >= _tokenAmounts[i]);
+            fulfillment.fulfillers[j].transfer(calculateFraction(_tokenAmounts[i], fulfillment.numerators[j], fulfillment.denomenator));
+          } else {
+            require(_payoutTokens[i].balanceOf(this) >= _tokenAmounts[i]);
+            require(_payoutTokens[i].transfer(fulfillment.fulfillers[j], calculateFraction(_tokenAmounts[i], fulfillment.numerators[j], fulfillment.denomenator)));
+          }
+        }
+      }
+
+      FulfillmentAccepted(fulfillments.length - 1);
+  }
+
+  function drainBounty(StandardToken[] _payoutTokens)
+      public
+      onlyController
+  {
+    for (uint256 i = 0; i < _payoutTokens.length; i++){
+      uint toPay;
+      if (_payoutTokens[i] == address(0x0)){
+        toPay = this.balance;
+        controller.transfer(toPay);
+
+      } else {
+        toPay = _payoutTokens[i].balanceOf(this);
+        require(_payoutTokens[i].transfer(controller, toPay));
+      }
+    }
+      BountyDrained(msg.sender);
+  }
+
+  function changeBounty(address _controller, string _data)
```

Comment:
Why do we have a general changebounty function that just affects data and controller?  Having each piece separated clearly makes things easier from a syncing perspective. That way we can bind to discreet events.

ie. ControllerChanged, DataChanged, etc. etc.

Reply:
We could write functions for changing just those as well, but combining them just prevents having to make 2 calls if you want to change both at once

Reply:
yeah makes sense. It's just a pain - and yeah it would be silly from a ui perspective if they have to authorize 2 transactions. In this case I'd probably want a different event.  The fact that Kauri is calling us via another contract was a huge game changer.

Reply:
put up issue on events.